OptimizelySDK.Tests/UtilsTests/EventTagUtilsTest.cs

```diff
@@ -0,0 +1,208 @@
+                { "revenue", 42 }
+            };
+            var validTag2 = new Dictionary<string, object>() {
+                { "revenue", 100 }
+            };
+
+            var invalidTag = new Dictionary<string, object>() {
+                { "abc", 42 }
+            };
+            var nullValue = new Dictionary<string, object>() {
+                { "revenue", null }
+            };
+            var invalidValue = new Dictionary<string, object>() {
+                { "revenue", 42.5 }
+            };
+
+            // Invalid data.
+            Assert.Null(EventTagUtils.GetRevenueValue(null));
+            Assert.Null(EventTagUtils.GetRevenueValue(invalidTag));
+            Assert.Null(EventTagUtils.GetRevenueValue(nullValue));
+            Assert.Null(EventTagUtils.GetRevenueValue(invalidValue));
+
+            // Valid data.
+            Assert.AreEqual(EventTagUtils.GetRevenueValue(validTag), expectedValue);
+            Assert.AreEqual(EventTagUtils.GetRevenueValue(validTag2), expectedValue2);
+        }
+
+        [Test]
+        public void TestGetEventValue()
+        {
+            int expectedValue = 42;
+            float expectedValue2 = 42.5F;
+            double expectedValue3 = 42.52;
+
+            var validTag = new Dictionary<string, object>() {
+                { "value", 42 }
+            };
+
+            var validTag2 = new Dictionary<string, object>() {
+                { "value", 42.5 }
+            };
+
+            var validTag3 = new Dictionary<string, object>() {
+                { "value", 42.52 }
+            };
+
+            var invalidTag = new Dictionary<string, object>() {
+                { "abc", 42 }
+            };
+            var nullValue = new Dictionary<string, object>() {
+                { "value", null }
+            };
+            var validTagStr = new Dictionary<string, object>() {
+                { "value", "42" }
+            };
+            var validTagStr1 = new Dictionary<string, object>() {
+                { "value", "42.3" }
+            };
+
+            // Invalid data.
+            Assert.Null(EventTagUtils.GetNumericValue(null));
+            Assert.Null(EventTagUtils.GetNumericValue(invalidTag));
+            Assert.Null(EventTagUtils.GetNumericValue(nullValue));
+
+
+            // Valid data.
+            Assert.AreEqual(42, EventTagUtils.GetNumericValue(validTagStr));
+            Assert.AreEqual("42.3", EventTagUtils.GetNumericValue(validTagStr1).ToString());
+            Assert.AreEqual(EventTagUtils.GetNumericValue(validTag), expectedValue);
+            Assert.AreEqual(EventTagUtils.GetNumericValue(validTag2), expectedValue2);
+            Assert.AreEqual(EventTagUtils.GetNumericValue(validTag3).ToString(), expectedValue3.ToString());
+        }
+
+        [Test]
+        public void TestGetNumericMetricInvalidArgs()
+        {
+            Assert.IsNull(EventTagUtils.GetNumericValue(null));
+
+            //Errors for all, because it accepts only dictionary// 
```

Comment:
If the format of the event tag value is invalid, you should catch the error and return null.

Reply:
it accepts only dictionary (strongly typed), otherwise it will give compilation error. if it was runtime issue, we were able to implement it.

Reply:
@msohailhussain -- please address this.
I'm assuming you'd get a compiler error if the parameter is not a dictionary? Please confirm.

Reply:
yes, if the parameter is not dictionary, compile time error will come.